cfgrammar/src/lib/yacc/ast.rs

```diff
@@ -202,8 +209,138 @@ impl GrammarAST {
                 spans: vec![Span::new(0, 0)],
             });
         }
+
+        for sym in &self.expect_unused {
+            match sym {
+                crate::yacc::ast::Symbol::Rule(sym_name, sym_span) => {
```

Comment:
This can be shortened to `Symbol::Rule`, or maybe `self::Symbol::` but not `ast::Symbol` because it *is* the `ast` module, and unfortunately something like `use self as ast;` isn't allowed.

I expect it is probably fine to just shorten it to `Symbol::Rule`, but during the first patches I was dealing with both `Symbol` types a bit.